Change type conversion from float/dfloat to real

tsihyoung opened the pull request:
Change type conversions in `slateratom/lib` from `float` or `dfloat` to `real(,dp)`.

aradi commented on README.rst:
```diff
@@ -23,7 +23,7 @@ Prerequisites
 * Python3
 
-* LibXC library with f90 interface (tested with version 4.3.4, version 5.x does
+* LibXC library with f90 interface (tested with version 5.1.6, version 4.x does
```
Note, at this stage the code still uses the old interface, this belongs to the other PR.

aradi commented on README.rst:
```diff
@@ -32,11 +32,15 @@ Building the code
   (e.g. ``_build``)::
 
-    FC=gfortran cmake -DCMAKE_INSTALL_PREFIX=$HOME/opt/skprogs -B _build .
+    FC=gfortran cmake -DCMAKE_INSTALL_PREFIX=$HOME/opt/skprogs -DCMAKE_Fortran_FLAGS=-fopenmp -B _build .
```
Please remove the `-fopenmp` flag. OpenMP support should be added via OpenMP-detection in CMake and linking against the `OpenMP::OpenMP_Fortran` target.

aradi commented on README.rst:
```diff
@@ -32,11 +32,15 @@ Building the code
 
-    FC=gfortran cmake -DCMAKE_INSTALL_PREFIX=$HOME/opt/skprogs -B _build .
+    FC=gfortran cmake -DCMAKE_INSTALL_PREFIX=$HOME/opt/skprogs -DCMAKE_Fortran_FLAGS=-fopenmp -B _build .
+
+  or::
+
+    FC=ifort cmake -DCMAKE_INSTALL_PREFIX=$HOME/opt/skprogs -DCMAKE_Fortran_FLAGS=-qopenmp -B _build .
```
I think, this 2nd example is superfluous. (SkProgs should work with any standard conforming compiler, gfortran is just an example.)

aradi commented on README.rst:
```diff
@@ -47,8 +51,14 @@ Follow the usual CMake build workflow:
     
     PKG_CONFIG_PATH=FOLDER_WITH_LIBXC_PC_FILES FC=gfortran cmake [...]
 
+  or::
+
+    CMAKE_PREFIX_PATH=YOUR_LIBXC_INSTALL_FOLDER FC=ifort cmake [...]
+    
+    PKG_CONFIG_PATH=FOLDER_WITH_LIBXC_PC_FILES FC=ifort cmake [...]
```
Again, this is superfluous. Also, the `CMAKE_PREFIX_PATH` example will only work if we upgrade to recent libxc, so that belongs into the other PR.

aradi commented on slateratom/lib/dft.f90:
```diff
@@ -65,8 +65,11 @@ subroutine density_grid(p,max_l,num_alpha,poly_order,alpha,num_mesh_points,&
     real(dp) :: rhotot,rhodiff,drhotot,ddrhotot,drhodiff,ddrhodiff
-    integer :: ii,jj,kk,ll,mm,nn,oo
-    type(xc_f90_pointer_t) :: xcfunc_x, xcfunc_c, xcinfo
+    integer :: ii,jj,kk,ll,mm,oo
+    integer(c_size_t) :: nn
+    !type(xc_f90_pointer_t) :: xcfunc_x, xcfunc_c, xcinfo
+    type(xc_f90_func_t) :: xcfunc_x, xcfunc_c
+    type(xc_f90_func_info_t) :: xcinfo
```
That would only work with libxc 5.x

aradi commented on slateratom/lib/dft.f90:
```diff
@@ -76,11 +79,15 @@ subroutine density_grid(p,max_l,num_alpha,poly_order,alpha,num_mesh_points,&
-      call xc_f90_func_init(xcfunc_c, xcinfo, XC_LDA_C_PW, XC_POLARIZED)
+      call xc_f90_func_init(xcfunc_x, XC_LDA_X, XC_POLARIZED)
+      xcinfo = xc_f90_func_get_info(xcfunc_x)
+      call xc_f90_func_init(xcfunc_c, XC_LDA_C_PW, XC_POLARIZED)
+      xcinfo = xc_f90_func_get_info(xcfunc_x)
     elseif (xcnr == 3) then
-      call xc_f90_func_init(xcfunc_x, xcinfo, XC_GGA_X_PBE, XC_POLARIZED)
-      call xc_f90_func_init(xcfunc_c, xcinfo, XC_GGA_C_PBE, XC_POLARIZED)
+      call xc_f90_func_init(xcfunc_x, XC_GGA_X_PBE, XC_POLARIZED)
+      xcinfo = xc_f90_func_get_info(xcfunc_x)
+      call xc_f90_func_init(xcfunc_c, XC_GGA_C_PBE, XC_POLARIZED)
+      xcinfo = xc_f90_func_get_info(xcfunc_x)
```
Belong into a separate PR which upgrades libxc

aradi reviewed (changes requested):
Thanks for the PR! I agree with all the changes `float` and `dfloat` to `real(dp)`. But please remove everything else (especially the libxc upgrade), that belongs into a separate PR.

tsihyoung commented:
> `at` to `real(dp)`. But please remove everything else (especially the libxc upgrade), that belongs into a separate PR.

Thanks for reviewing this PR. I have removed the libxc interface changes and modified the README file.
